lib/BackbeatConsumer.js

```diff
@@ -797,7 +820,7 @@ class BackbeatConsumer extends EventEmitter {
                 if (this._zookeeper) {
                     this._zookeeper.close();
                 }
-                if (this._consumer) {
+                if (this._consumer && this._consumer.isConnected()) {
```

Comment:
when is this happening?

Reply:
This is linked to your PR, when a rebalance times out and the consumer gets disconnected we don't want in that case to wait for the disconnect event.